sdk/forms/src/protected-view/iframe/components/view.ts

```diff
@@ -2,57 +2,91 @@ import { component } from '../../../common/component';
-});
+export const View = component(
+  ({ result, display, sendToParent }: { result: Result; display: DisplayOptions; sendToParent: Sender }) => {
+    const view = document.createElement('div');
+    view.classList.add('view');
+    const value = result.strategy === 'invoke-action' ? result.response : result.objects;
+    const values = display.map(({ path, label, clickToCopy, class: className }) =>
+      DisplayValue({ value, path, label, clickToCopy, className, sendToParent }),
+    );
+    view.replaceChildren(...values);
+    return view;
+  },
+);
 
-type DisplayValueProps = { path?: string; value: unknown; label?: string; clickToCopy?: boolean; className?: string };
-
-const DisplayValue = component(({ value: rootValue, label, path, className, clickToCopy }: DisplayValueProps): Node => {
-  const container = document.createElement('div');
-  if (className) container.className = className;
-  if (path) container.setAttribute('data-path', path);
-
-  const value = followPath(rootValue, path);
-  let children: Node[] = [];
-  if (label) children.push(Label({ label }));
-
-  switch (typeof value) {
-    case 'string':
-    case 'number':
-    case 'boolean': {
-      const span = document.createElement('span');
-      span.innerText = String(value);
-      container.appendChild(span);
-      children.push(span);
-      break;
-    }
-    case 'object':
-      if (value === null) break;
+type DisplayValueProps = {
+  sendToParent: Sender;
+  path?: string;
+  value: unknown;
+  label?: string;
+  clickToCopy?: boolean;
+  className?: string;
+};
+
+const DisplayValue = component(
+  ({ value: rootValue, label, path, className, clickToCopy, sendToParent }: DisplayValueProps): Node => {
+    const container = document.createElement('div');
+    if (className) container.className = className;
+    if (path) container.setAttribute('data-path', path);
+
+    const value = followPath(rootValue, path);
+    const children: Node[] = [];
+    if (label) children.push(Label({ label }));
 
-      if (Array.isArray(value)) {
-        children.push(...value.map((item, index) => DisplayValue({ value: item, clickToCopy })));
+    switch (typeof value) {
+      case 'string':
+      case 'number':
+      case 'boolean': {
+        const span = document.createElement('span');
+        span.innerText = String(value);
+        container.appendChild(span);
+        children.push(span);
+        span.addEventListener('mouseenter', (event) => {
+          sendToParent('mouseenter', {
+            path,
+            x: event.clientX,
+            y: event.clientY,
+          });
+        });
+        span.addEventListener('mouseleave', () => {
+          sendToParent('mouseleave', {
+            path,
+          });
+        });
+        span.addEventListener('click', () => {
+          if (clickToCopy) {
+            navigator.clipboard.writeText(String(value));
+          }
+          sendToParent('click', {
+            path,
+          });
+        });
```

Comment:
We probably need to call `removeEventListener` at some point if the element is unmounted.
Let's extend the `component` function to support cleanup logic.

Reply:
Done, added optional `unmount` prop to the `renderer` response type, which will be responsible for cleanup logic

Reply:
Done